### PR TITLE
feat: internal access modifier

### LINKED ISSUE #4156: `internal` access modifier
### Feature Spec

`internal` access modifier can be used anywhere the `pub` access modifier can be applied (on type definitions, methods, fields, etc.).
When a type or method or other API element has the `internal` access modifier, it behaves like it's public (`pub`) within the current Wing library or module, but it will appear private when the library or module is imported by other Wing source code.

### Use Cases

```wing
class A {
  internal static do_a() {};
}
class B {
  public static do_b() {
    A.do_a(); // We don't want access to `do_a` outside this file, but we do need access to it in this exported public method
  }
}
```

### Implementation Notes

_No response_

### Component

Language Design, Compiler

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by Chriscbr**
Closes #4156

It's been a while since I've worked with the symbol lookup subsystem. This area is in dire need of refactoring, unfortunately the system is a little too broad to take on in this PR (I'm also not sure what the immediate fix is off the top of my head). Some of the issues:
- It's a bit confusing which `lookup_xxx` method should be called in different scenarios
- The logic for handling on access modifiers errors is split between two places (once inside `type_check.rs` for determining if field/method accesses are valid, and once inside `symbol_env.rs` for determining if looking up types within environments are valid)
- The logic also needs to be added/maintained separately for the LSP.

There are also some edge cases I intentionally avoided getting side tracked with to focus on just getting the initial feature working. For example, there might be cases we should look towards other languages for how to handle, like in what situations is it okay to subclass a method as internal.

I did some manual QA to validate that autocompletions work as expected with the language server.

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: tools/hangar/__snapshots__/invalid.ts.snap [hunk 1]
```diff
@@ -2843,6 +2843,104 @@ Test Files 1 failed (1)
 Duration <DURATION>"
 `;
 
+exports[`internal_access_modifiers.test.w 1`] = `
+"error: Expected identifier "fixture" to be a variable, but it's a namespace
+  --> ../../../examples/tests/invalid/internal_access_modifiers.test.w:3:9
+  |
+3 | let e = fixture.FavoritePlanets.MARS;
+  |         ^^^^^^^
```
Comment (Chriscbr) [inline tools/hangar/__snapshots__/invalid.ts.snap:2851]: Confusing error message if you try to access an internal enum

File: tools/hangar/__snapshots__/invalid.ts.snap [hunk 2]
```diff
@@ -2843,6 +2843,104 @@ Test Files 1 failed (1)
...
+   |
+34 | internal interface InternalInterface {}
+   |                    ----------------- defined here
+
+
+error: Expected an interface, instead found type "unresolved"
+  --> ../../../examples/tests/invalid/internal_access_modifiers.test.w:6:22
+  |
+6 | class FakeClass impl fixture.InternalInterface {}
+  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (Chriscbr) [inline tools/hangar/__snapshots__/invalid.ts.snap:2870]: Unneeded extra error when you extend an internal interface

**Review by eladb — APPROVED** [review]
Horray!

File: tools/hangar/__snapshots__/invalid.ts.snap (hunk 1 above)
Comment (Chriscbr) [reply tools/hangar/__snapshots__/invalid.ts.snap:2851]: https://github.com/winglang/wing/issues/6985

File: tools/hangar/__snapshots__/invalid.ts.snap (hunk 2 above)
Comment (Chriscbr) [reply tools/hangar/__snapshots__/invalid.ts.snap:2872]: https://github.com/winglang/wing/issues/6986

**Pull request merged**